core/src/main/java/org/wso2/testgrid/core/command/GenerateTestPlanCommand.java

```diff
@@ -498,12 +505,38 @@ private Yaml createYamlInstance() {
                 Properties configAwareInfraCombination = toConfigAwareInfrastructureCombination(
                         combination.getParameters());
                 testPlan.getInfrastructureConfig().setParameters(configAwareInfraCombination);
+
+                ScenarioConfig scenarioConfig = testgridYaml.getScenarioConfig();
+                List<ConfigChangeSet> configChangeSets = scenarioConfig.getConfigChangeSets();
+                if (configChangeSets != null) {
+                    List<TestScenario> modifiedTestScenarios = new ArrayList<>();
+                    for (ConfigChangeSet configChangeSet : configChangeSets) {
+                        for (TestScenario testScenario : scenarioConfig.getScenarios()) {
+                             TestScenario newTestScenario = new TestScenario();
```

Comment:
Since the array is calling 'modifiedTestScenarios', I think `modifiedTestScenario` would be a better name, WDYT?